src/main/scala/mesosphere/marathon/api/akkahttp/Directives.scala

```diff
@@ -13,6 +20,64 @@ import scala.concurrent.duration._
   * These should be imported by the respective controllers
   */
 object Directives extends AuthDirectives with LeaderDirectives with AkkaDirectives {
+  /**
+    * Matches the rest of the path segment as a PathId; ignores trailing slash, consumes everything.
+    */
+  object RemainingPathId extends PathMatcher1[PathId] {
+    import akka.http.scaladsl.server.PathMatcher._
+
+    @tailrec final def iter(reversePieces: List[String], remaining: Path): Matching[Tuple1[PathId]] = remaining match {
```

Comment:
What is meant by `reversePieces`?

Reply:
Should it be renamed to `reversedSegments`?